test/cpp/util/tls_test_utils.h

```diff
@@ -76,6 +76,25 @@ class AsyncCertificateVerifier
   std::deque<Request> queue_ ABSL_GUARDED_BY(mu_);
 };
 
+class VerifiedRootCertSubjectVerifier
+    : public grpc::experimental::ExternalCertificateVerifier {
+ public:
+  explicit VerifiedRootCertSubjectVerifier(const std::string& expected_subject)
+      : expected_subject_(expected_subject) {}
```

Comment:
When I add `std::move()` here I get a clang-tidy error `std::move of const variable has no effect`

Reply:
Option 1: The way this is currently written:
- If the caller has a `std::string` object, then we'll get a ref to that object passed into this ctor, and we'll make a copy of the `std::string` for the new object.
- If the caller has a `char*` instead of a `std::string`, then the compiler will construct a temporary `std::string` object from the `char*` in the caller, so that it has an object that it can pass down a reference to, then we'll make a copy of that `std::string` for the new object, and then the compiler will immediately destroy the temporary `std::string` it created in the caller.  So we'll create two `std::string` objects and then immediately delete one of them.
- If the caller has an `absl::string_view`, the caller will need to explicitly convert it to a `std::string`.  The result in this case is basically the same as the `char*` case, but it's a little more annoying to write, because there is no explicit conversion from `absl::string_view` to `std::string`.

Option 2: If you change the parameter to `std::string` and use `std::move()` here:
- If the caller has a `std::string` object, then a copy will be made to pass the string in by value, and then we'll move that copy into place in the new object, so we're just creating a single `std::string`.
- If the caller has a `char*`, the compiler will construct a new `std::string` object from the `char*`, it will pass that in by value, and we'll move it into place in the new object, so we're just creating a single `std::string`.
- If the caller has an `absl::string_view`, the caller will need to explicitly convert it to `std::string`.  The result in this case is basically the same as the `char*` case, but it's a little more annoying to write, because there is no explicit conversion from `absl::string_view` to `std::string`.
- If the caller happens to have a `std::string` that they don't need to hold on to, then the caller can use `std::move()` to pass in the parameter, in which case we're not creating *any* new `std::string` objects.  (This case isn't necessarily that common, but it can be a useful optimization at times.)

Option 3: Another option is to use `absl::string_view` as the parameter type, in which case:
- Regardless of whether the caller has a `std::string`, `char*`, or `absl::string_view`, we will construct a new `absl::string_view` to pass in by value (which is much cheaper than constructing a `std::string`).  We will then create a new `std::string` from the `absl::string_view` for this object, so we're creating just one `std::string`.

In general, `absl::string_view` is probably the right thing to use for any C++ code that takes a string as input.  But using `std::string` and then using `std::move()` to store it does allow optimizing for the occasional case where the caller already has a `std::string` that they no longer need.  So either of those options seems like a reasonable choice here.

To be clear, the performance does not matter at all in this particular case, but I find that it's generally a good idea to consider the performance implications of the various ways of writing this sort of thing, so that you're in good habits when you do encounter a case where it really matters. :)

Reply:
Thanks for the thorough explanation! I've swapped the parameter to an `absl::string_view`